### PR TITLE
fix(proxy-cache): keep cache_method same with nginx's proxy_cache_methods

### PR DESCRIPTION
**Pull request by agile6v**
### What this PR does / why we need it:

Nginx's [proxy_cache_methods](https://github.com/nginx/nginx/blob/926e0aa70acd58d2a13451f40fc19aeb5393aa7e/src/http/ngx_http_upstream.c#L456
) directive only allows to specify `GET`、`POST` and `HEAD`, so we keep same with Nginx.

#4776 

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by spacewander — APPROVED** [review]

File: apisix/cli/ngx_tpl.lua [hunk 1]
```diff
@@ -596,7 +596,7 @@ http {
             proxy_cache                         $upstream_cache_zone;
             proxy_cache_valid                   any {% if proxy_cache.cache_ttl then %} {* proxy_cache.cache_ttl *} {% else %} 10s {% end %};
             proxy_cache_min_uses                1;
-            proxy_cache_methods                 GET HEAD;
+            proxy_cache_methods                 GET HEAD POST;
```
Comment (tokers) [inline apisix/cli/ngx_tpl.lua:599]: Nginx by default doesn't enable the cache for POST. Why change this behavior here?

Comment (agile6v) [reply apisix/cli/ngx_tpl.lua:599]: It doesn't affect the default behavior of APISIX, which means that user can specify the `POST` method on APISIX.

Comment (tzssangglass) [reply apisix/cli/ngx_tpl.lua:599]: Is it necessary to cache a POST request?

The only I can think of for caching POST requests is that some internal specification forces the consistent use of POST requests……

Comment (tokers) [reply apisix/cli/ngx_tpl.lua:599]: > It doesn't affect the default behavior of APISIX, which means that user can specify the `POST` method on APISIX.

Here you change the Nginx template, without a configurable setting. This is a change for default havior.

Comment (agile6v) [reply apisix/cli/ngx_tpl.lua:599]: > @agile6v
> Could you sync it to APISIX.pm?

done

Comment (agile6v) [reply apisix/cli/ngx_tpl.lua:599]: > Is it necessary to cache a POST request?
> 
> The only I can think of for caching POST requests is that some internal specification forces the consistent use of POST requests……

Yes. This is a very rare scenario. The users don't enable the method on APISIX,  it doesn't take effect.  If users enable it, they need to know what they are doing.

Comment (agile6v) [reply apisix/cli/ngx_tpl.lua:599]: > > It doesn't affect the default behavior of APISIX, which means that user can specify the `POST` method on APISIX.
> 
> Here you change the Nginx template, without a configurable setting. This is a change for default havior.

Whether to enable is [here](https://github.com/agile6v/apisix/blob/master/docs/en/latest/plugins/proxy-cache.md) (cache_method)

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**